Tests/PublicAPISpec.swift

```diff
@@ -342,6 +338,44 @@ class PublicAPISpec: QuickSpec {
                     expect(UIApplication.shared.keyWindow?.subviews)
                         .toAfterTimeout(allPass({ !($0 is BaseView) }))
                 }
+
+                it("will transfer cached data (sync) from anonymous user") {
+                    messageMixerService.mockedResponse = TestHelpers.MockResponse.withGeneratedCampaigns(
+                        count: 1, test: false, delay: 100, maxImpressions: 2, addContexts: false,
+                        triggers: [Trigger(type: .event,
+                                           eventType: .loginSuccessful,
+                                           eventName: "e1",
+                                           attributes: [])])
+                    RInAppMessaging.registerPreference(nil)
+                    RInAppMessaging.logEvent(LoginSuccessfulEvent())
+                    expect(UIApplication.shared.keyWindow?.subviews).toEventually(containElementSatisfying({
+                        $0 is BaseView
+                    })) // wait
+
+                    RInAppMessaging.registerPreference(IAMPreferenceBuilder().setUserId("user").build())
+                    let identifiers = [UserIdentifier(type: .userId, identifier: "user")]
+                    expect(dataCache.getUserData(identifiers: identifiers)?.campaignData?.first?.impressionsLeft)
+                        .toEventually(equal(1))
+                }
+
+                it("will not transfer cached data (sync) from empty user") {
```

Comment:
I guess this makes sense(?)

Reply:
Can empty user happen in a normal flow?

Reply:
In DemoApp yes but in general case I think that's not expected behaviour. On the other hand I'm not sure if we should threat empty user as anonymous (nil) user. I think it's a product level decision